src/k2/module/k23si/Module.cpp

```diff
@@ -749,13 +754,17 @@ K23SIPartitionModule::handleRead(dto::K23SIReadRequest&& request, FastDeadline d
         return _makeReadOK(rec);
     }
 
+    count++;
     // record is still pending and isn't from same transaction.
     return _doPush(request.key, versions.WI->data.timestamp, request.mtr, deadline)
```

Comment:
I think this overall logic should be done in one place - inside _doPush itself. I'd rather pass this counter in there  and generate the correct response before attempting actual work.

That way it also works across the board and you don't have to duplicate this code for read/write/query

Reply:
good suggestion, moved the check logic to _doPush().